### PR TITLE
[FLINK-15359] Remove unused YarnConfigOptions, Tests, Docs

### PR DESCRIPTION
**Pull request by MGloder**
## What is the purpose of the change

This PR delete unused YarnConfigOptions code to avoid misunderstanding

## Brief change log

Delete unused code from YarnConfigOptions, ConfigConstants and YARNSessionCapacitySchedulerITCase. Also regenerate the docs for unused options

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency):  no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no 
  - The runtime per-record code paths (performance sensitive): no 
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Comment by TisonKun** [comment]
@xintongsong possibly the "maximum failed containers" is still useful(but no power). Please also take a look if it is a bug we should fix or a outdated configuration we can remove.

**Comment by xintongsong** [comment]
@TisonKun 
I'm not aware of any reason for keeping the "maximum failed containers".
I checked the git history, and it seems this option was last used by the legacy `YarnFlinkResourceManager`, which is removed in `e97967278904cb2706a6514ad0a13d50cc389b20`. The commit was actually authored by you and committed by @tillrohrmann.
Maybe @tillrohrmann could help double check on this option?

**Comment by TisonKun** [comment]
@xintongsong Yes it is ever used in pre FLIP-6 framework. If we keep current dynamically allocate container mechanism it possibly meaningless. We can always add back the configuration when we implement similar logic in the future but given it has no power now remove them could reduce misleading.

File: flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java [hunk 1]
```diff
@@ -434,17 +434,6 @@
...
-	 * The maximum number of failed YARN containers before entirely stopping
-	 * the YARN session / job on YARN.
-	 *
-	 * <p>By default, we take the number of initially requested containers.
-	 *
-	 * @deprecated in favor of {@code YarnConfigOptions#MAX_FAILED_CONTAINERS}.
-	 */
-	@Deprecated
-	public static final String YARN_MAX_FAILED_CONTAINERS = "yarn.maximum-failed-containers";
-
```
Comment (tillrohrmann) [inline flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: You cannot simply remove static fields from a class which is annotated with `@Public`.

Comment (TisonKun) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: OK. Will push a revert commit. Why the CI system give green to the change. IIRC japi checks are integrated.

Comment (TisonKun) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: Thanks for the explanation. I will push a commit to revert the removal of `ConfigConstants#YARN_MAX_FAILED_CONTAINERS` and be careful changes in `@Public`.

Besides, I'd like to know the reason we turn off japicmp plugin for `ConfigConstants`.

Comment (TisonKun) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: Another thought is that per @aljoscha mentioned before remove an option with no power notify the user about a problematic settings. But well, here the concern is b/w promise provided by `@Public`

Comment (tillrohrmann) [reply flink-core/src/main/java/org/apache/flink/configuration/ConfigConstants.java]: Turning off the japicmp plugin check for `ConfigConstants` was a mistake.

One could introduce a special `ConfigOption` which throws an exception if being used. That would have the same effect as breaking a program build after upgrading.

**Pull request merged**